src/app/clusters/time-synchronization-server/time-synchronization-delegate.h

```diff
@@ -104,12 +104,17 @@ class Delegate
      * a CHIP_ERROR.
      */
     virtual CHIP_ERROR UpdateTimeUsingNTPFallback(const CharSpan & fallbackNTP,
-                                                  chip::Callback::Callback<OnFallbackNTPCompletion> * callback) = 0;
+                                                  chip::Callback::Callback<OnFallbackNTPCompletion> * callback)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
 
     /**
      * @brief Signals application that UTCTime has changed through the timesync cluster.
      */
-    virtual void UTCTimeAvailabilityChanged(uint64_t time) = 0;
+    virtual void UTCTimeAvailabilityChanged(uint64_t time) {}
```

Comment:
I definitely like that this is in here now, but the delegate is now kind of a mix of service requests and notifications. There's likely only one place where we should be asking for time updates, but there are potentially LOTS of things that might want these notifications. Would it make sense to separate the platform delegate from a notifications class that can be registered by the apps?

Reply:
That sounds like a good idea. This was added only to address the need for apps to get time when it becomes available. We can do the platform level additions in a separate effort.